File: backend/StudioManagement.Data/Entities/Expense.cs
namespace StudioManagement.Data.Entities;

public class Expense : ITenantEntity
{
    public int ExpenseId { get; set; }
    public int StudioId { get; set; }
    public int ExpenseCategoryId { get; set; }
    public int? EventId { get; set; }
    public int? WorkerId { get; set; }
    public DateTime ExpenseDate { get; set; }
    public decimal Amount { get; set; }
    public string? Description { get; set; }
    public string? PaymentMethod { get; set; }
    public string? ReferenceNumber { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public byte[] RowVersion { get; set; } = null!;

    public Studio Studio { get; set; } = null!;
    public ExpenseCategory ExpenseCategory { get; set; } = null!;
    public Event? Event { get; set; }
    public Worker? Worker { get; set; }
}

File: frontend/src/types/expense.ts
export interface Expense {
  expenseId: number;
  expenseCategoryId: number;
  expenseCategoryName: string;
  eventId: number | null;
  eventVenue: string | null;
  workerId: number | null;
  workerName: string | null;
  expenseDate: string;
  amount: number;
  description: string | null;
  paymentMethod: string | null;
  referenceNumber: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateExpenseRequest {
  expenseCategoryId: number;
  eventId?: number;
  workerId?: number;
  expenseDate: string;
  amount: number;
  description?: string;
  paymentMethod?: string;
  referenceNumber?: string;
}

export type UpdateExpenseRequest = CreateExpenseRequest;
